src/explore-education-statistics-admin/src/pages/release/edit-release/content/ReleaseContext.tsx

```diff
@@ -0,0 +1,237 @@
+import produce from 'immer';
+import {
+  EditableContentBlock,
+  ExtendedComment,
+  EditableRelease,
+} from '@admin/services/publicationService';
+import { DataBlock } from '@common/services/dataBlockService';
+import { ContentSection } from '@common/services/publicationService';
+import React, { createContext, useContext, useReducer } from 'react';
+import ReleaseDispatchAction from './actions';
+
+type Dispatch = (action: ReleaseDispatchAction) => void;
+type State = {
+  release: EditableRelease | undefined;
+  canUpdateRelease: boolean;
+  availableDataBlocks: DataBlock[];
+  unresolvedComments: ExtendedComment[];
+};
+type ReleaseProviderProps = { children: React.ReactNode };
+
+const ReleaseStateContext = createContext<State | undefined>(undefined);
+const ReleaseDispatchContext = createContext<Dispatch | undefined>(undefined);
+
+export function releaseReducer(state: State, action: ReleaseDispatchAction) {
```

Comment:
If we use `useImmerReducer`, we can type `releaseReducer` against `use-immer`'s `Reducer` type which allows us to write this as:

```javascript
const releaseReducer: Reducer<ReleaseContextState, ReleaseContextAction> = (draft, action) => {
  //...
}
```